python/pylibraft/pylibraft/test/test_kmeans.py

```diff
@@ -16,11 +16,40 @@
 import numpy as np
 import pytest
 
-from pylibraft.cluster.kmeans import cluster_cost, compute_new_centroids
+from pylibraft.cluster.kmeans import (
+    KMeansParams,
+    cluster_cost,
+    compute_new_centroids,
+    fit,
+)
 from pylibraft.common import Handle, device_ndarray
 from pylibraft.distance import pairwise_distance
 
 
+@pytest.mark.parametrize("n_rows", [100])
+@pytest.mark.parametrize("n_cols", [5, 25])
+@pytest.mark.parametrize("n_clusters", [5, 15])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+def test_kmeans_fit(n_rows, n_cols, n_clusters, dtype):
+    # generate some random input points / centroids
+    X_host = np.random.random_sample((n_rows, n_cols)).astype(dtype)
+    centroids = device_ndarray(X_host[:n_clusters])
+    X = device_ndarray(X_host)
+
+    # compute the inertia, before fitting centroids
+    original_inertia = cluster_cost(X, centroids)
+
+    # TODO: test out some different options on this
+    # TODO: use a fixed RNG state on params
+    params = KMeansParams(n_clusters=n_clusters)
+
+    # fit the centroids, make sure inertia has gone down
```

Comment:
Can you create a github issue to expose make_blobs in a future PR and reference it here in this comment? 

Ultimately, the make_blobs allows us to generate Gaussian clusters that we can directly compare the output against. I think it's out of scope for this PR but we definitely want to revisit.

Reply:
Created https://github.com/rapidsai/raft/issues/1059 - and added a TODO in the test